onnxruntime/core/optimizer/gemm_sum_fusion.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/optimizer/rewrite_rule.h"
+
+namespace onnxruntime {
+
+/**
+@Class GemmTransposeFusion
+
+Rewrite rule that fuses Gemm and Sum nodes to a single Gemm node.
+This fusion can be applied in the following scenario:
+1) Sum at output of Gemm: when the output of a Gemm is immedietly summed with
+    exactly one other element, we can fuse this Sum with Gemm by using the other
+    Sum input as C, provided that the C input to the Gemm is missing.
+    is set accordingly. This is supported for opset >= 11, as this is when Gemm
+    input C became optional.
+
+TODO: Support the Add use case: Sum(x, y) ~= Add.
+ 
+It is attempted to be triggered only on nodes with op type "Gemm".
+
+A --> Gemm --> Sum   ==  A -----+
+       ^        ^               |
+       |        |               v
+B -----+        C        B --> Gemm <-- C
```

Comment:
Let's show all tensor names in the graph and make statement clearer. There is one tensor missed between `Gemm` and `Sum`. Consider

```
A --> Gemm -->D -->  Sum  --> E 
        ^              ^                                          
        |              |                                       
B ------'              C
```
 is equivalent to   

```
A --> Gemm --> E 
      ^ ^                                    
      | |                                       
B ----' C                                      
```
where a single alphabet means a tensor.